api-example.xml

```diff
@@ -0,0 +1,129 @@
+<problem display_name="webGLDemo">
+<script type="loncapa/python">
+<![CDATA[
+import json
+def vglcfn(e, ans):
+  '''
+  Grading a force diagram.
+  '''
+
+  ### Helpers
+  def compare_lengths(v1, v2):
+      return abs(v1.length - v2.length) < 1.00
+
+  def check_angle(v1, expected, tolerance=1.0):
+      return abs(v1.angle - expected) < tolerance
+
+  ### Checks ###
+
+  def check_all_vectors_present(answer):
+      vectors = answer['vectors']
+      if not ('N' in vectors and 'g' in vectors and 'f' in vectors):
+          return 'You need to use all three vectors.'
+
+  def check_compare_normal_and_gravity_lengths(answer):
+      normal = answer['vectors']['N']
+      gravity = answer['vectors']['g']
+      if not compare_lengths(normal, gravity):
+          return 'Normal Force and Gravitational Force should be similar lengths.'
+
+  def check_normal_angle(answer):
+      normal = answer['vectors']['N']
+      if not check_angle(normal, 90, tolerance=2.0):
+          return 'The angle of the Normal Force is incorrect. Your angle: %.1f' % normal.angle
+
+  def check_gravity_angle(answer):
+      gravity = answer['vectors']['g']
+      if not check_angle(gravity, 270.0, tolerance=2.0):
+          return 'The angle of the Gravitational Force is incorrect. Your angle: %.1f' % gravity.angle
+
+  def check_friction_angle(answer):
+      friction = answer['vectors']['f']
+      if not check_angle(friction, 180.0, tolerance=2.0):
+          return 'The angle of the Frictional Force is incorrect. Your angle: %.1f' % friction.angle
+
+  def check_starts_at_center_of_mass(answer, tolerance=0.25):
+      cm = answer['points']['cm']
+      for vec in answer['vectors'].values():
+          dist = math.hypot(cm.x - vec.tail.x, cm.y - vec.tail.y)
+          if dist > tolerance:
+              return 'One or more of your vectors do not start from the center of mass.'
+
+  checks = [
+      check_all_vectors_present,
+      check_compare_normal_and_gravity_lengths,
+      check_normal_angle,
+      check_gravity_angle,
+      check_friction_angle,
+      check_starts_at_center_of_mass,
+  ]
+  success_message = 'Good job!'
+
+
+  ### Python API - Do not edit after this line ####
```

Comment:
@mtyaka nit: it might make sense to separate Python API into a dedicated file. The idea is to have it copy-pasted into every vectordraw problem, so if we could have a clear separation of API and what's built on top of it it would be great.